app/services/journey_logger_service.rb

```diff
@@ -0,0 +1,31 @@
+class JourneyLoggerService
+  class << self
+    def call(assessment_id, calculation_result, estimate)
+      attributes = build_attributes(calculation_result, estimate)
+      CompletedUserJourney.transaction do
+        if (journey = CompletedUserJourney.find_by(assessment_id:))
+          journey.update!(attributes)
+        else
+          CompletedUserJourney.create!(attributes.merge(assessment_id:))
+        end
+      end
+    rescue StandardError => e
+      ErrorService.call(e)
+    end
+
+    def build_attributes(calculation_result, estimate)
+      {
+        certificated: !estimate.controlled?,
+        partner: estimate.partner || false,
+        person_over_60: estimate.over_60 || (estimate.partner && estimate.partner_over_60) || false,
```

Comment:
could we just have `estimate&.partner_over_60` in the first OR section?

Reply:
and the same with `partner_vehicle_owned?`

Reply:
Annoyingly no - because of the edge case where the user has said the client has a partner over 60 then gone back and said the client doesn't have a partner after all. In that case there is no person over 60, even though estimate.partner_over_60 will be `true`.